include/ginkgo/core/base/executor.hpp

```diff
@@ -1607,6 +1612,14 @@ class CudaExecutor : public detail::ExecutorBase<CudaExecutor>,
         return this->get_exec_info().max_subgroup_size;
     }
 
+    /**
+     * Get maximum shared memory per block.
+     */
+    int get_max_shared_memory_per_block() const noexcept
```

Comment:
does this function need to be part of the executor? It could just as well be implemented just in the cuda/hip backend

Reply:
I think this could be useful in other kernels as well.

Reply:
It's not useful for core right now. If we come to the situation where it is needed for core, we can revisit this. But since we are quite rushing all of these PRs, I think it's good to remove any interface change that is not absolutely necessary.

Reply:
now moved to internal

Reply:
thanks!